go/tools/gopackagesdriver/bazel_json_builder.go

```diff
@@ -166,7 +167,24 @@ func (b *BazelJSONBuilder) Build(ctx context.Context, mode LoadMode) ([]string,
 		"--aspects=" + strings.Join(aspects, ","),
 		"--output_groups=" + b.outputGroupsForMode(mode),
 		"--keep_going", // Build all possible packages
-	}, bazelBuildFlags, labels)
+	}, bazelBuildFlags)
+
+	if len(labels) < 100 {
+		buildArgs = append(buildArgs, labels...)
+	} else {
+		// To avoid hitting MAX_ARGS length, write labels to a file and use `--target_pattern_file`
+		targetsFile, err := ioutil.TempFile("", "gopackagesdriver_targets_")
+		if err != nil {
+			return nil, fmt.Errorf("unable to create target pattern file: %w", err)
+		}
+		targetsFile.WriteString(strings.Join(labels, "\n"))
```

Comment:
Can you write to the file line by line instead? It would have much smaller memory footprint

Reply:
Done